WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsDetailListFragment.kt

```diff
@@ -0,0 +1,629 @@
+        listView.divider = null
+        listView.dividerHeight = 0
+        listView.setHeaderDividersEnabled(false)
+        if (footerView != null) {
+            listView.addFooterView(footerView)
+        }
+        reloadNoteBlocks()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        setUniqueIdToView(listView)
+        if (activity is ScrollableViewInitializedListener) {
+            (activity as ScrollableViewInitializedListener).onScrollableViewInitialized(listView.id)
+        }
+
+        // Set the note if we retrieved the noteId from savedInstanceState
+        if (!TextUtils.isEmpty(restoredNoteId)) {
+            setNote(restoredNoteId)
+            reloadNoteBlocks()
+            restoredNoteId = null
+        }
+        if (note == null) {
+            showErrorToastAndFinish()
+        }
+    }
+
+    override fun onPause() {
+        // Stop the reader comment service if it is running
+        ReaderCommentService.stopService(activity)
+        restoreOriginalViewId(listView)
+        super.onPause()
+    }
+
+    override fun getNote(): Note? {
+        return notification
+    }
+
+    override fun setNote(noteId: String?) {
+        if (noteId == null) {
+            showErrorToastAndFinish()
+            return
+        }
+        val note: Note? = NotificationsTable.getNoteById(noteId)
+        if (note == null) {
+            showErrorToastAndFinish()
+            return
+        }
+        notification = note
+    }
+
+    private fun showErrorToastAndFinish() {
+        AppLog.e(NOTIFS, "Note could not be found.")
+        activity?.let {
+            ToastUtils.showToast(activity, string.error_notification_open)
+            it.finish()
+        }
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        notification?.let {
+            outState.putString(KEY_NOTE_ID, it.id)
+            outState.putInt(KEY_LIST_POSITION, listView.firstVisiblePosition)
+        }
+
+        super.onSaveInstanceState(outState)
+    }
+
+    private fun reloadNoteBlocks() {
+        LoadNoteBlocksTask().executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR)
+    }
+
+    fun setFooterView(footerView: ViewGroup?) {
+        this.footerView = footerView
+    }
+
+    private val mOnNoteBlockTextClickListener: OnNoteBlockTextClickListener = object : OnNoteBlockTextClickListener {
+        override fun onNoteBlockTextClicked(clickedSpan: NoteBlockClickableSpan) {
+            if (!isAdded || activity !is NotificationsDetailActivity) {
+                return
+            }
+            handleNoteBlockSpanClick(activity as NotificationsDetailActivity, clickedSpan)
+        }
+
+        override fun showDetailForNoteIds() {
+            if (!isAdded || notification == null || activity !is NotificationsDetailActivity) {
+                return
+            }
+            val detailActivity = activity as NotificationsDetailActivity
+
+            requireNotNull(notification).let { note ->
```

Comment:
`requireNotNull` check is not present in Java file, and I think we might not need it here, since we already check if `notification == null`.

I see this pattern couple of times in code.

Reply:
AFAIU being those mutable member variables, we need at least to assert them as not null and for example assign them to a local variable even after the check previously done; something like:

```
val localNote : Note = notification!!
```

The usage of the requireNotNull seemed to me a bit more synthetic maybe 🤷 ; not feeling strong about, or if I'm missing anything so let me know wdyt 😄

Reply:
Ah, right, it's not only about null 👍 `requireNotNull` is fine with me :)